addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc1/NikoHomeControlCommunication1.java

```diff
@@ -0,0 +1,542 @@
+            } else if ("listactions".equals(cmd)) {
+                cmdListActions(((NhcMessageListMap1) nhcMessageGson).getData());
+            } else if (("listthermostat").equals(cmd)) {
+                cmdListThermostat(((NhcMessageListMap1) nhcMessageGson).getData());
+            } else if ("executeactions".equals(cmd)) {
+                cmdExecuteActions(((NhcMessageMap1) nhcMessageGson).getData());
+            } else if ("executethermostat".equals(cmd)) {
+                cmdExecuteThermostat(((NhcMessageMap1) nhcMessageGson).getData());
+            } else if ("listactions".equals(event)) {
+                eventListActions(((NhcMessageListMap1) nhcMessageGson).getData());
+            } else if ("listthermostat".equals(event)) {
+                eventListThermostat(((NhcMessageListMap1) nhcMessageGson).getData());
+            } else if ("getalarms".equals(event)) {
+                eventGetAlarms(((NhcMessageMap1) nhcMessageGson).getData());
+            } else {
+                logger.debug("Niko Home Control: not acted on json {}", nhcMessage);
+            }
+        } catch (JsonParseException e) {
+            logger.debug("Niko Home Control: not acted on unsupported json {}", nhcMessage);
+        }
+    }
+
+    private synchronized void cmdSystemInfo(Map<String, String> data) {
+        logger.debug("Niko Home Control: systeminfo");
+
+        if (data.containsKey("swversion")) {
+            this.systemInfo.setSwVersion(data.get("swversion"));
+        }
+        if (data.containsKey("api")) {
+            this.systemInfo.setApi(data.get("api"));
+        }
+        if (data.containsKey("time")) {
+            this.systemInfo.setTime(data.get("time"));
+        }
+        if (data.containsKey("language")) {
+            this.systemInfo.setLanguage(data.get("language"));
+        }
+        if (data.containsKey("currency")) {
+            this.systemInfo.setCurrency(data.get("currency"));
+        }
+        if (data.containsKey("units")) {
+            this.systemInfo.setUnits(data.get("units"));
+        }
+        if (data.containsKey("DST")) {
+            this.systemInfo.setDst(data.get("DST"));
+        }
+        if (data.containsKey("TZ")) {
+            this.systemInfo.setTz(data.get("TZ"));
+        }
+        if (data.containsKey("lastenergyerase")) {
+            this.systemInfo.setLastEnergyErase(data.get("lastenergyerase"));
+        }
+        if (data.containsKey("lastconfig")) {
+            this.systemInfo.setLastConfig(data.get("lastconfig"));
+        }
+    }
+
+    /**
+     * Return the object with system info as read from the Niko Home Control controller.
+     *
+     * @return the systemInfo
+     */
+    public synchronized NhcSystemInfo1 getSystemInfo() {
+        return this.systemInfo;
+    }
+
+    private void cmdStartEvents(Map<String, String> data) {
+        Integer errorCode = Integer.valueOf(data.get("error"));
```

Comment:
There is some mixed usage of primitive `int` and object `Integer` in the class.  On some other places it also uses the object only be assigned to a `primitive` and here it uses an object to only compare to 0. Better use `parseInt` as it just returns a `int`. So the code can just use `int`.
```suggestion
        int errorCode = Integer.parseInt(data.get("error"));
```

Reply:
Revisited all my usage of Integer vs int.